jdbc-driver/src/test/java/com/google/cloud/dataproc/jdbc/DataprocDriverUtilsTest.java

```diff
@@ -0,0 +1,174 @@
+    public void testHiveUrlSimpleWithDbAndPort() throws IOException, InvalidURLException {
+        String simpleUrlWithDbAndPort =
+                "jdbc:dataproc://hive/db-name;port=10003;clusterName=simple-cluster";
+        HiveJdbcConnectionParam param = DataprocDriverUtils.parseHiveUrl(simpleUrlWithDbAndPort);
+        assertThat(param.host()).isEqualTo(HOST);
+        assertThat(param.port()).isEqualTo(PORT);
+        assertThat(param.dbName()).isEqualTo(DB_NAME);
+
+        assertThat(param.transportMode()).isEqualTo(simpleParam.transportMode());
+        assertThat(param.httpPath()).isEqualTo(simpleParam.httpPath());
+        assertThat(param.otherSessionConfs()).isEqualTo(simpleParam.otherSessionConfs());
+        assertThat(param.hiveConfs()).isEqualTo(simpleParam.hiveConfs());
+        assertThat(param.hiveVars()).isEqualTo(simpleParam.hiveVars());
+    }
+
+    @Test
+    public void testHiveUrlLongWithOtherSessionConfs() throws IOException, InvalidURLException {
+        String longUrlWithOtherSessionConfs =
+                "jdbc:dataproc://hive/db-name;port=10003;clusterName=simple-cluster;user=foo;password=bar";
+        String hiveUrl =
+                "jdbc:hive2://simple-cluster-m:10003/db-name;transportMode=http;httpPath=cliservice;user=foo;password=bar";
+        HiveJdbcConnectionParam param = DataprocDriverUtils.parseHiveUrl(longUrlWithOtherSessionConfs);
+
+        assertThat(hiveUrl).isEqualTo(param.toHiveJdbcUrl());
+        assertThat(param.host()).isEqualTo(HOST);
+        assertThat(param.dbName()).isEqualTo(DB_NAME);
+        assertThat(param.port()).isEqualTo(PORT);
+        assertThat(param.transportMode()).isEqualTo(TRANSPORT_MODE);
+        assertThat(param.httpPath()).isEqualTo(HTTP_PATH);
+        assertThat(param.otherSessionConfs()).isEqualTo(";user=foo;password=bar");
+        assertThat(param.hiveConfs()).isNull();
+        assertThat(param.hiveVars()).isNull();
+    }
+
+    @Test
+    public void testHiveUrlLongWithHiveConfs() throws IOException, InvalidURLException {
+        String longUrlWithHiveConfs =
+                "jdbc:dataproc://hive/db-name;port=10003;clusterName=simple-cluster?hive.support.concurrency=true";
+        String hiveUrl =
+                "jdbc:hive2://simple-cluster-m:10003/db-name;transportMode=http;httpPath=cliservice?hive.support.concurrency=true";
+        HiveJdbcConnectionParam param = DataprocDriverUtils.parseHiveUrl(longUrlWithHiveConfs);
+
+        assertThat(hiveUrl).isEqualTo(param.toHiveJdbcUrl());
+        assertThat(param.host()).isEqualTo(HOST);
+        assertThat(param.dbName()).isEqualTo(DB_NAME);
+        assertThat(param.port()).isEqualTo(PORT);
+        assertThat(param.transportMode()).isEqualTo(TRANSPORT_MODE);
+        assertThat(param.httpPath()).isEqualTo(HTTP_PATH);
+        assertThat(param.otherSessionConfs()).isNull();
+        assertThat(param.hiveConfs()).isEqualTo("hive.support.concurrency=true");
+        assertThat(param.hiveVars()).isNull();
```

Comment:
It could be better to assert by constructing expected instance of `HiveJdbcConnectionParam` and asserting if it equals to `param`.